### DISCUSSION TITLE
What does "AutoPlay will work fully autonomously" mean?

### DISCUSSION THREAD

**Discussion by lucascompython** [EvieePy/Wavelink#272] [General]
What does "AutoPlay will work fully autonomously" mean? 
What does `partial` do?

Also, if it is enabled it fills the auto_queue, but then if when I get a track (`player.queue.get()`) it will use the auto_queue if the queue is empty.
I understand that this may be intentional, but it's not clear in the docs.

I've also found that if I had a recommended track playing and then manually add a track, when the track ends, `on_wavelink_track_end` will be called twice.

And the `queue` and `auto_queue` attributes are missing from [Player](https://wavelink.dev/en/latest/wavelink.html#player), in the docs.

**Comment by EvieePy** [comment]
AutoPlay is **not** supposed to be used in conjunction with manually manipulating the way the player plays songs at the end of previous ones.

Autonomously means, independent of what the user does, AutoPlay will continue to do it's own thing. Which is why it is recommended to **not** use `on_wavelink_track_end` when you have AutoPlay set to either `partial` or `enabled`. If you use this event, you **should** return early if AutoPlay is set to `partial` or `enabled`.

`.partial` is simply gathering the next song in your queue (if there is one) and playing it with the addition of some error handling/logic.

`.enabled` will fetch recommendations for you based on previous listening habits this session and add them to a separate queue (`auto_queue`) and play from this continuously. If you add a song to your normal queue, as stated in the docs: `If a song is put into a players standard queue, AutoPlay will use it as a priority.`.

`on_wavelink_track_end` should not be called twice and if it is, it's probably because you are doing something in this event to conflict with AutoPlay when it is `enabled` or `partial`.

I did not notice these two attributes missing from the docs, thanks for the pickup!

**Reply by EvieePy** [reply]
Also see: https://wavelink.dev/en/latest/migrating.html#autoplay

**Reply by EvieePy** [reply]
And: https://wavelink.dev/en/latest/wavelink.html#wavelink.on_wavelink_track_end

It is usually recommended to not do logic in this event when AutoPlay is on `partial` or `enabled`.

**Comment by lucascompython** [comment]
Oh, I didn't know that autoplay would literally autoplay songs in the queue.
I got it working nicely now, thank you for the support!
